agent_identity/src/service/aggregate.rs

```diff
@@ -88,120 +82,128 @@ impl Aggregate for Service {
-                    .origin(origin.clone())
-                    .issuance_date(issuance_date)
-                    .expiration_date(expiration_date)
-                    .build()
-                    .map_err(|err| DomainLinkageCredentialBuilderError(err.to_string()))?
-                    .serialize_jwt(Default::default())
-                    .map_err(|err| SerializationError(err.to_string()))?;
-
-                // Compose JWT
-                let header = Header {
-                    alg: get_preferred_signing_algorithm(),
-                    typ: None,
-                    // TODO: make dynamic
-                    kid: Some(format!("{subject_did}#key-0")),
-                    ..Default::default()
-                };
+                let mut linked_dids = vec![];
+
+                // For each Verification Method, create a new linked DID JWT token.
+                for verification_method in verification_methods {
+                    let subject_did = verification_method.id().did();
+
+                    let verification_method_id = verification_method.id();
+                    let alg = verification_method
+                        .data()
+                        .public_key_jwk()
+                        .and_then(|jwk| jwk.alg())
+                        .ok_or_else(|| MissingVerificationMethodAlgorithm(verification_method_id.to_string()))?;
+                    let algorithm = Algorithm::from_str(alg)
+                        .map_err(|_| UnsupportedVerificationMethodAlgorithm(alg.to_string()))?;
+
+                    let domain_linkage_credential = DomainLinkageCredentialBuilder::new()
+                        .issuer(subject_did.clone())
+                        .origin(origin.clone())
+                        .issuance_date(issuance_date)
+                        .expiration_date(expiration_date)
+                        .build()
+                        .map_err(|err| DomainLinkageCredentialBuilderError(err.to_string()))?
+                        .serialize_jwt(Default::default())
+                        .map_err(|err| SerializationError(err.to_string()))?;
+
+                    // Compose JWT
+                    let header = Header {
+                        alg: algorithm,
+                        typ: None,
+                        kid: Some(verification_method.id().to_string()),
+                        ..Default::default()
+                    };
+
+                    let linked_did = [
+                        URL_SAFE_NO_PAD.encode(
+                            header
+                                .to_json_vec()
+                                .map_err(|err| SerializationError(err.to_string()))?,
+                        ),
+                        URL_SAFE_NO_PAD.encode(domain_linkage_credential.as_bytes()),
+                    ]
+                    .join(".");
+
+                    let proof_value = subject
+                        .sign(linked_did.as_str(), "placeholder", algorithm)
```

Comment:
Is this `"placeholder"` supposed to stay in there? If yes, please document where it comes from, what that is or mark it with a `TODO:` if it is intended to be changed at some point.

Reply:
I added a TODO comment :ok_hand: